### PR TITLE
fix: Switching between multiple locations

### LINKED ISSUE #934: bug: Switching between multiple locations, does not change the url
## Bug Definition
When there are multiple pins on same location there is a swap button at the bottom of the drawer.
With these buttons switching between locations does not update the current url.

-- Please add discord contact information
** discord username: tinman#7219 **

## Bug environment
rc.afetharita.com
afetharita.com

## Describe how you are producing the bug step by step

https://rc.afetharita.com/en?lat=37.076078833957666&lng=36.24636143445969&zoom=18&id=552626
Screenshot 2023-02-12 at 9 41 24 PM

Click on the pin shown in the image, switch to next pin in the drawer.

## Expected Behaviour
Should update the url.

## Desktop Information
- Operating System: osx
- Browser any

### PR DESCRIPTION
**Pull request by muratyilmaz743**
# Description

**discord username: murat#5479**

**closes #934 

Please describe your changes. Also describe your aim and content. Do not forget to list the dependencies required caused by those changes.

## Things to check before creating a PR
- [x] I have inspected my topic, checked.
- [ ] If it is a core feature, executed detailed tests.

## Creating PR rules
- [ ] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [ ] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [ ] Clean commit history: The commits where the changes are made should be clear and organized.
- [ ] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [ ] A new feature (a change that adds a new feature, not a breaking change)
- [x] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

# How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: src/components/UI/Drawer/components/OtherRecordsInSameLocation.tsx [hunk 1]
```diff
@@ -26,14 +27,33 @@ type Props = {
...
+        pathname: "",
+        query: {
+          ...router.query, // list all the queries here
+          id: reference,
+        },
+      },
+      undefined,
+      {
+        shallow: true,
+      }
```
Comment (eraygundogmus) [inline src/components/UI/Drawer/components/OtherRecordsInSameLocation.tsx:52]: what is all of these and why?

**Comment by absolutezero13** [comment]
this is fixed by #964 which is merged @eraygundogmus fyi

**Comment by eraygundogmus** [comment]
related #964

**Comment by muratyilmaz743** [comment]
I think there may be a confusion, #964 merged for the problem that the id parameter is not removed after the closed drawer but this pr created for Switching between multiple locations, does not update the url #934 which is still open. @eraygundogmus @absolutezero13

File: src/components/UI/Drawer/components/OtherRecordsInSameLocation.tsx (hunk 1 above)
Comment (muratyilmaz743) [reply src/components/UI/Drawer/components/OtherRecordsInSameLocation.tsx:52]: to update the url after Switching between multiple locations

**Comment by absolutezero13** [comment]
@muratyilmaz743 It also solves that issue as well, basically listening drawerData change to update url. You can check the deployment on the PR.

**Comment by muratyilmaz743** [comment]
> @muratyilmaz743 It also solves that issue as well, basically listening drawerData change to update url. You can check the deployment on the PR.

Okay then, thank you for your support.

**Pull request closed without merging**